File: designPatterns/Behavioral_patterns/Memento/javascript/Memento.js
/**
 * The Memento interface provides a way to retrieve the memento's metadata, such
 * as creation date or name. However, it doesn't expose the Originator's state.
 */
class Memento {
  getState() {}

  getName() {}

  getDate() {}
}

module.exports = Memento;

File: designPatterns/Behavioral_patterns/Memento/python/Memento.py
from abc import ABC, abstractmethod


class Memento(ABC):
    """
    The Memento interface provides a way to retrieve the memento's metadata,
    such as creation date or name. However, it doesn't expose the Originator's
    state.
    """

    @abstractmethod
    def get_name(self):
        pass

    @abstractmethod
    def get_date(self):
        pass
